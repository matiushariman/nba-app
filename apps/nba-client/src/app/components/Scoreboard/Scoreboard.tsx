import ScoreboardCard from './ScoreboardCard';
import useGetScoreboard from '../../hooks/useGetScoreboard';

const Scoreboard = () => {
  const { data, isLoading } = useGetScoreboard();

  return (
    <div className="container mx-auto">
      <div className="grid md:pt-6 md:grid-cols-2 gap-6">
        {isLoading
          ? null
          : data?.scoreboard.games.map((game) => (
              <ScoreboardCard
                key={game.gameId}
                homeTeam={game.homeTeam}
                awayTeam={game.awayTeam}
                gameStatusText={game.gameStatusText}
              />
            ))}
      </div>
    </div>
  );
};

export default Scoreboard;
