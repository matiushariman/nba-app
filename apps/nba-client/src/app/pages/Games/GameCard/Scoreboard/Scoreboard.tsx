import { useNavigate } from 'react-router-dom';

import TeamDetails from './TeamDetails';
import Score from './Score';
import GameStatusText from '../../../../components/GameStatusTextWithContext';

import type { ScoreboardProps } from './Scoreboard.types';

const Scoreboard = ({ awayTeam, homeTeam, gameId }: ScoreboardProps) => {
  const navigate = useNavigate();

  const navigateToGameDetails = () => navigate(`game-details/${gameId}`);

  return (
    <div className="flex-grow flex flex-col divide-y">
      <div className="pt-2 pb-2 pl-4 pr-4 flex gap-2 items-center flex-grow">
        <div className="flex gap-3 items-center">
          <TeamDetails
            teamId={awayTeam.teamId}
            teamName={awayTeam.teamName}
            wins={awayTeam.wins}
            losses={awayTeam.losses}
          />
          <Score score={awayTeam.score} />
        </div>
        <div className="flex-grow flex justify-center">
          <GameStatusText />
        </div>
        <div className="flex gap-3 items-center">
          <Score score={homeTeam.score} />
          <TeamDetails
            teamId={homeTeam.teamId}
            teamName={homeTeam.teamName}
            wins={homeTeam.wins}
            losses={homeTeam.losses}
          />
        </div>
      </div>
      <div
        onClick={navigateToGameDetails}
        role="button"
        className="hidden text-center flex items-center justify-center pt-2 pb-2 pl-4 pr-4 uppercase text-sm font-bold text-gray-800 bg-gray-50"
      >
        Game Details
      </div>
    </div>
  );
};

export default Scoreboard;
