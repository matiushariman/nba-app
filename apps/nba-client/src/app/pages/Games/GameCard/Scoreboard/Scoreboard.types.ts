import type { GameCardProps } from '../GameCard.types';

export type ScoreboardProps = Omit<GameCardProps, 'gameLeaders'>;
