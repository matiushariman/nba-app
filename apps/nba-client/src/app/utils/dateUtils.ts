import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
