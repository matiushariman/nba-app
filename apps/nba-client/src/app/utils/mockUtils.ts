export const shouldUseMock = process.env['NODE_ENV'] === 'development';
