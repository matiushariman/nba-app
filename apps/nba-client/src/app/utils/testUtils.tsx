import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
