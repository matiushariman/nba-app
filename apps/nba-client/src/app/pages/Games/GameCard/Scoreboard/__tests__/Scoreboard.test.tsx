import Scoreboard from '..';
import { render } from '../../../../../utils/testUtils';

describe('pages/Games/GameCard/Scoreboard', () => {
  const homeTeam = {
    teamId: 0,
    teamName: 'Celtics',
    wins: 82,
    losses: 0,
    score: 131,
  };
  const awayTeam = {
    teamId: 1,
    teamName: 'Lakers',
    wins: 0,
    losses: 82,
    score: 92,
  };

  it('should render Scoreboard that matches snapshot', () => {
    const { baseElement } = render(
      <Scoreboard
        homeTeam={homeTeam}
        awayTeam={awayTeam}
        gameStatusText="FINAL"
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
