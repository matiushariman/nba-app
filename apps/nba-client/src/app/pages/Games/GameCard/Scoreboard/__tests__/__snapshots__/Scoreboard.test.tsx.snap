// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Games/GameCard/Scoreboard should render Scoreboard that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="pt-2 pb-2 pl-4 pr-4 flex gap-2 items-center"
    >
      <div
        class="flex gap-3 items-center"
      >
        <div
          class="grid gap-1 justify-center"
        >
          <img
            alt="Lakers logo"
            height="64"
            loading="lazy"
            src="https://cdn.nba.com/logos/nba/1/global/L/logo.svg"
            width="64"
          />
          <div>
            <p
              class="text-center text-sm"
            >
              Lakers
            </p>
            <p
              class="text-center text-gray-500 text-xs"
            >
              0 - 82
            </p>
          </div>
        </div>
        <div
          class="w-[52px]"
        >
          <p
            class="font-bold text-2xl"
          >
            92
          </p>
        </div>
      </div>
      <div
        class="flex-grow flex justify-center"
      >
        <p
          class="text-sm uppercase font-bold"
        >
          FINAL
        </p>
      </div>
      <div
        class="flex gap-3 items-center"
      >
        <div
          class="w-[52px]"
        >
          <p
            class="font-bold text-2xl"
          >
            131
          </p>
        </div>
        <div
          class="grid gap-1 justify-center"
        >
          <img
            alt="Celtics logo"
            height="64"
            loading="lazy"
            src="https://cdn.nba.com/logos/nba/0/global/L/logo.svg"
            width="64"
          />
          <div>
            <p
              class="text-center text-sm"
            >
              Celtics
            </p>
            <p
              class="text-center text-gray-500 text-xs"
            >
              82 - 0
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
